openpype/modules/deadline/plugins/publish/submit_houdini_render_deadline.py

```diff
@@ -27,138 +35,99 @@ class HoudiniSubmitRenderDeadline(pyblish.api.InstancePlugin):
-                key=key,
-                value=environment[key]
-            ) for index, key in enumerate(environment)
-        })
-
-        # Include OutputFilename entries
-        # The first entry also enables double-click to preview rendered
-        # frames from Deadline Monitor
-        output_data = {}
         for i, filepath in enumerate(instance.data["files"]):
             dirname = os.path.dirname(filepath)
             fname = os.path.basename(filepath)
-            output_data["OutputDirectory%d" % i] = dirname.replace("\\", "/")
-            output_data["OutputFilename%d" % i] = fname
+            job_info.OutputDirectory = dirname.replace("\\", "/")
+            job_info.OutputFilename = fname
 
-            # For now ensure destination folder exists otherwise HUSK
-            # will fail to render the output image. This is supposedly fixed
-            # in new production builds of Houdini
-            # TODO Remove this workaround with Houdini 18.0.391+
-            if not os.path.exists(dirname):
-                self.log.info("Ensuring output directory exists: %s" %
-                              dirname)
-                os.makedirs(dirname)
+        return job_info
 
-        payload["JobInfo"].update(output_data)
+    def get_plugin_info(self):
 
-        self.submit(instance, payload)
-
-    def submit(self, instance, payload):
+        instance = self._instance
+        context = instance.context
 
-        AVALON_DEADLINE = legacy_io.Session.get("AVALON_DEADLINE",
-                                          "http://localhost:8082")
-        assert AVALON_DEADLINE, "Requires AVALON_DEADLINE"
+        # Output driver to render
+        driver = instance[0]
+        hou_major_minor = hou.applicationVersionString().rsplit(".", 1)[0]
 
-        plugin = payload["JobInfo"]["Plugin"]
-        self.log.info("Using Render Plugin : {}".format(plugin))
+        plugin_info = DeadlinePluginInfo(
+            SceneFile=context.data["currentFile"],
+            OutputDriver=driver.path(),
+            Version=hou_major_minor,
+            IgnoreInputs=True
+        )
 
-        self.log.info("Submitting..")
-        self.log.debug(json.dumps(payload, indent=4, sort_keys=True))
+        return attr.asdict(plugin_info)
 
-        # E.g. http://192.168.0.1:8082/api/jobs
-        url = "{}/api/jobs".format(AVALON_DEADLINE)
-        response = requests.post(url, json=payload)
-        if not response.ok:
-            raise Exception(response.text)
+    def process(self, instance):
+        super(HoudiniSubmitDeadline, self).process(instance)
 
+        # TODO: Avoid the need for this logic here, needed for submit publish
         # Store output dir for unified publisher (filesequence)
         output_dir = os.path.dirname(instance.data["files"][0])
         instance.data["outputDir"] = output_dir
-        instance.data["deadlineSubmissionJob"] = response.json()
+        instance.data["toBeRenderedOn"] = "deadline"
```

Comment:
@antirotor I've refactored it now to use the AbstractSubmitDeadline base class.

However, without setting this instance data the Submit Publish Job will fail with errors.
Where should these be set instead of in this plug-in? And, are they even necessary at all - in particular `toBeRenderedOn`?

Reply:
This is remant of shared code for submitting jobs to deadline and muster. Since muster part isn't maintained an royal render isn't written yet, this is quite unnecessary and should be cleaned from the code anyway.

Reply:
I'm not sure whether to do that in this PR or leave it 'ugly' like this until this PR and do a cleanup pass in a separate PR for this comment and the other comment I posted below. @antirotor @mkolar Thoughts?